### PR TITLE
Add `nonfungible_position_manager`

### PR DESCRIPTION
**Pull request by shuhuiluo**

### CODE REVIEW

**Review by malik672 — APPROVED** [review]

**Comment by malik672** [comment]
failed the linter

File: src/nonfungible_position_manager.rs [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+            WETH9::new().get(1).unwrap().clone(),
+            FeeAmount::MEDIUM,
+            encode_sqrt_ratio_x96(1, 1),
+            0,
+            None,
+        )
+        .unwrap()
+    }
+    const RECIPIENT: Address = address!("0000000000000000000000000000000000000003");
+    const SENDER: Address = address!("0000000000000000000000000000000000000004");
```
Comment (malik672) [inline src/nonfungible_position_manager.rs:445]: according to clippy this is not used, are you like saving it for later use, it's a test anyway not like it has any impact on performance in real time

File: src/nonfungible_position_manager.rs [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+                .abi_encode(),
+            );
+        }
+    }
+
+    let mut value = U256::ZERO;
+
+    if let Some(ether) = options.use_native {
+        let wrapped = ether.wrapped();
+        assert!(
```
Comment (malik672) [inline src/nonfungible_position_manager.rs:198]: feel like it's better to recover the err here and i don't think it's good to use assertion in prod or runtime(might be me though)

**Review by malik672 — APPROVED** [review]

File: src/nonfungible_position_manager.rs (hunk 2 above)
Comment (shuhuiluo) [reply src/nonfungible_position_manager.rs:198]: Will do that later.

**Pull request merged**